direct/src/showbase/ShowBase.py

```diff
@@ -512,41 +517,41 @@ def __init__(self, fStartDirect=True, windowType=None):
 
         # DO NOT ADD TO THIS LIST.  We're trying to phase out the use of
         # built-in variables by ShowBase.  Use a Global module if necessary.
-        builtins.base = self
-        builtins.render2d = self.render2d
-        builtins.aspect2d = self.aspect2d
-        builtins.pixel2d = self.pixel2d
-        builtins.render = self.render
-        builtins.hidden = self.hidden
-        builtins.camera = self.camera
-        builtins.loader = self.loader
-        builtins.taskMgr = self.taskMgr
-        builtins.jobMgr = self.jobMgr
-        builtins.eventMgr = self.eventMgr
-        builtins.messenger = self.messenger
-        builtins.bboard = self.bboard
+        builtins.base = self  # type: ignore[attr-defined]
+        builtins.render2d = self.render2d  # type: ignore[attr-defined]
+        builtins.aspect2d = self.aspect2d  # type: ignore[attr-defined]
+        builtins.pixel2d = self.pixel2d  # type: ignore[attr-defined]
+        builtins.render = self.render  # type: ignore[attr-defined]
+        builtins.hidden = self.hidden  # type: ignore[attr-defined]
+        builtins.camera = self.camera  # type: ignore[attr-defined]
+        builtins.loader = self.loader  # type: ignore[attr-defined]
+        builtins.taskMgr = self.taskMgr  # type: ignore[attr-defined]
+        builtins.jobMgr = self.jobMgr  # type: ignore[attr-defined]
+        builtins.eventMgr = self.eventMgr  # type: ignore[attr-defined]
+        builtins.messenger = self.messenger  # type: ignore[attr-defined]
+        builtins.bboard = self.bboard  # type: ignore[attr-defined]
```

Comment:
Hmm, this is awkward, but I guess there's no other way... I'm told that we'd need a custom version of typeshed otherwise to be able to define custom builtins.  Do these ignores work with mypyc?

Reply:
It looks like, in general ,mypyc can still compile code like this, but it crashes if any of the values are accessed without the `builtins` prefix. If we want to have less `type: ignore` comments, we could add a function that sets attributes on `builtins`, so the `type: ignore` would only need to be in one place. Configuration could also switch the function to a no-op.

There might also be a way to set things up such that, when compiling with mypyc, some other object that does allow these attributes to be set is instead imported as `builtins`.

Reply:
I think if mypyc crashes when these are accessed, we shouldn't just ignore these... maybe we ought to just suppress attr-defined warnings?